mythtv/libs/libmythmetadata/metadatafactory.cpp

```diff
@@ -654,6 +658,17 @@ LookupType GuessLookupType(ProgramInfo *pginfo)
     if (ret != kUnknownVideo)
         return ret;
 
+    // The recording type may indicate that it's a television show.
+    // There is GetRecordingRuleType() in pginfo, but since the
+    // recording operation is finished, it has been set to
+    // kNotRecording by this time. We'll get the actual
+    // recording type from the RECORD table in the database.
+    RecordingType rt = pginfo->QueryRecordRuleType();
```

Comment:
Thanks for pointing that out. I didn't realize that the record type information I needed was so easily available. I'll rework the code to use the info from rule->Load(), and eliminate the QueryRecordRuleType() routine which I had added.

If the record type is **Daily** or **Weekly**, then I don't think it could be a movie. In most cases the **All** record type would similarly suggest a TV show, but you have pointed out one scenario in which it might still be a movie. I could declare probable television only for **Daily** or **Weekly**. I'm not thrilled with that because I have a rule generated from EIT where I ask it to record **All** episodes of "The Andy Griffith Show", and I'd like it to guess TV show.

In the kSingleRecord, kOneRecord, or kAllRecord cases, we have nothing to base the decision on. That's why I wanted to consider the length of the recording to give us a hint. "The Andy Griffith Show" is broadcast in 30 minute length, for example. But the length criteria is fallible with unusually short movies or long TV shows. If it's right 90% of the time, is it worth keeping when we're trying to _**guess**_ the type?

The original code said that if it doesn't have a season, an episode, or a subtitle, then it must be a Movie. That's ridiculous. EIT info has no specified season, episode or subtitle where I live. Manual Record rules generally have none of these either. 

Should we return kUnknownVideo for cases where we're not sure?